### PR TITLE
Shooter pid

### PR DESCRIPTION
**Pull request by romallan**
-Runs feeder when its almost at setpoint (around 150 off)
-ends command when its 100 off setpoint

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/ShooterSubsystem.java [hunk 1]
```diff
@@ -76,4 +78,8 @@ public void set(double speed) {
     shooterMotorL.set(-speed);
     shooterMotorR.set(-speed);
   }
+
+  public boolean atGoal() {
+    return getMeasurement() >= goal - 100;
```
Comment (minhd-vu) [inline src/main/java/frc/robot/subsystems/ShooterSubsystem.java]:
```suggestion
    return Math.abs(getMeasurement() - goal) <= 100;
```

File: src/main/java/frc/robot/subsystems/ShooterSubsystem.java [hunk 2]
```diff
@@ -50,6 +51,7 @@ public void periodic() {
   @Override
   public void useOutput(double output, double setpoint) {
     // Use the output here
+    goal= setpoint;
```
Comment (minhd-vu) [inline src/main/java/frc/robot/subsystems/ShooterSubsystem.java]:
```suggestion
    goal = setpoint;
```

File: src/main/java/frc/robot/subsystems/ShooterSubsystem.java (hunk 1 above)
Comment (minhd-vu) [reply src/main/java/frc/robot/subsystems/ShooterSubsystem.java]: Make the threshold a constant

File: src/main/java/frc/robot/subsystems/ShooterSubsystem.java [hunk 3]
```diff
@@ -18,6 +18,7 @@ public class ShooterSubsystem extends PIDSubsystem {
   
   CANSparkFlex shooterMotorR;
   CANSparkFlex shooterMotorL;
+  double goal;
```
Comment (minhd-vu) [inline src/main/java/frc/robot/subsystems/ShooterSubsystem.java]:
```suggestion
 private double goal;
```

File: src/main/java/frc/robot/commands/manipulator/Shooter.java [hunk 4]
```diff
@@ -21,22 +20,21 @@ public Shooter(double shootSpeed) {
...
     RobotContainer.shooterSubsystem.enable();
-    commandTimer.restart();
   }
 
   // Called every time the scheduler runs while the command is scheduled.
   @Override
   public void execute() {
     RobotContainer.shooterSubsystem.setSetpoint(shootSpeed);
-    if (commandTimer.get() > TimeConstants.SpeakerFeed) {
+    if (RobotContainer.shooterSubsystem.getMeasurement() >= shootSpeed - 150) {
```
Comment (minhd-vu) [inline src/main/java/frc/robot/commands/manipulator/Shooter.java]: use at goal, but i don't think this will run the feeder for enough time

**Review by minhd-vu — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/manipulator/Shooter.java [hunk 5]
```diff
@@ -21,22 +20,21 @@ public Shooter(double shootSpeed) {
...
     RobotContainer.shooterSubsystem.enable();
-    commandTimer.restart();
   }
 
   // Called every time the scheduler runs while the command is scheduled.
   @Override
   public void execute() {
     RobotContainer.shooterSubsystem.setSetpoint(shootSpeed);
-    if (commandTimer.get() > TimeConstants.SpeakerFeed) {
+    if (Math.abs(RobotContainer.shooterSubsystem.getMeasurement()) >= Math.abs(shootSpeed - 150)) {
```
Comment (zabzzz) [inline src/main/java/frc/robot/commands/manipulator/Shooter.java]:
```suggestion
    if (RobotContainer.shooterSubsystem.atGoal() {

**Pull request closed without merging**